components/cdc/src/delegate.rs

```diff
@@ -614,19 +614,20 @@ impl Delegate {
             rows.push(v);
         }
         self.sink_downstream(rows, index, ChangeDataRequestKvApi::TiDb)?;
-        self.sink_raw_downstream(raw_rows, index)
+        self.sink_downstream(raw_rows, index, ChangeDataRequestKvApi::RawKv)
     }
 
-    fn sink_raw_downstream(&mut self, entries: Vec<EventRow>, index: u64) -> Result<()> {
-        if entries.is_empty() {
-            return Ok(());
+    pub fn raw_untrack_ts(&mut self, cdc_id: ObserveID, max_ts: TimeStamp) {
+        // Stale CmdBatch, drop it silently.
+        if cdc_id != self.handle.id {
+            return;
         }
         // the entry's timestamp is non-decreasing, the last has the max ts.
-        let max_raw_ts = TimeStamp::from(entries.last().unwrap().commit_ts);
+        // use prev ts, see reason at CausalObserver::pre_propose_query
+        let max_raw_ts = max_ts.prev();
         match self.resolver {
             Some(ref mut resolver) => {
-                // use prev ts, see reason at CausalObserver::pre_propose_query
-                resolver.raw_untrack_lock(max_raw_ts.prev());
+                resolver.raw_untrack_lock(max_raw_ts);
```

Comment:
Suggest to try adding unit test for the condition that `self.resolver` is none.

Reply:
added test case in `test_raw_pending_lock`